lib/src/wiggle_abi/dictionary_impl.rs

```diff
@@ -4,12 +4,32 @@ use {
     crate::{
         error::Error,
         session::Session,
-        wiggle_abi::{fastly_dictionary::FastlyDictionary, types::DictionaryHandle},
+        wiggle_abi::{
+            fastly_dictionary::FastlyDictionary,
+            types::{DictionaryHandle, FastlyStatus},
+        },
     },
     std::{convert::TryFrom, fs, path::Path},
     wiggle::GuestPtr,
 };
 
+#[derive(Debug, thiserror::Error)]
+pub enum DictionaryError {
+    /// A dictionary item with the given key was not found.
+    #[error("Unknown dictionary item")]
+    UnknownDictionaryItem,
```

Comment:
Quick question: what's the rationale for dropping the string paylaod here? Viceroy prints out the errors prior to converting them to ABI codes, and it seems helpful to provide that info.